app/javascript/schools/courses/components/course_editor/CourseEditor__Form.res

```diff
@@ -221,7 +227,18 @@ let createCourse = (state, send, reloadCoursesCB) => {
 let updateCourse = (state, send, updateCourseCB, course) => {
   send(StartSaving)
 
-  let updateCourseQuery = UpdateCourseQuery.make(
+  let highlights = Js.Array.map(
+    h =>
+      UpdateCourseQuery.makeInputObjectCourseHighlightInput(
+        ~title=Course.Highlight.title(h),
+        ~icon=Course.Highlight.icon(h),
+        ~description=Course.Highlight.description(h),
+        (),
+      ),
```

Comment:
`186-190` has the same lines, except for a small change. Old method has less duplication. :(

Reply:
I don't think we have any other way around when the input is another complex type ? 🤔

Reply:
I don't think we have any other way around when the input is another complex type ? 🤔